partner_tlr_update/models/res_partner.py

```diff
@@ -165,23 +161,35 @@ def parse_xml_tjansteleverantor_data(self, xml):
         root = etree.XML(xml.encode())
         self.update_from_xml(root, 'organization')
         _logger.info("UPDATE FROM XML")
-        contact_persons = root.findall(
-            ".//ns300:tjansteleverantor/ns107:kontaktpersonLista", 
-                namespaces={'ns300': "http://arbetsformedlingen.se/tjansteleverantor/response/hamtatjansteleverantorsvar/v15",
-                'ns107': 'http://arbetsformedlingen.se/datatyp/tjansteleverantor/tjansteleverantor/v15'})
-        if contact_persons:
-            _logger.info("CONTACT PERSONS NOT NONE")
-            for elem in contact_persons:
-                self.update_contact_person(elem, self.id, False)
         subsidiaries = root.findall(
             ".//ns107:utforandeVerksamhetLista", namespaces={'ns107': 'http://arbetsformedlingen.se/datatyp/tjansteleverantor/tjansteleverantor/v15'})
         if subsidiaries:
             _logger.info("SUBSIDIARY NOT NONE")
+            contract_nrs = self.get_contract_nrs(root)
             for elem in subsidiaries:
-                self.update_subsidiary(elem)
+                self.update_subsidiary(elem, contract_nrs)
 
     @api.multi
-    def update_contact_person(self, xml, parent_id, address_id):
+    def get_contract_nrs(self, xml, service_nr = 26):
+        contract_nrs = []
+        contracts = xml.findall(".//ns107:avtalLista", namespaces={'ns107': "http://arbetsformedlingen.se/datatyp/tjansteleverantor/tjansteleverantor/v15"})
+        if contracts:
+            _logger.info("contracts found")
+            for contract in contracts:
+                service_id = contract.find(".//ns62:tjanstId", namespaces={'ns62':'http://arbetsformedlingen.se/datatyp/tjansteleverantor/avtal/v15'})
+                if service_id is not None:
+                    _logger.info("service number %s found, service number wanted: %s" % (service_id.text,service_nr))
+                    if service_id.text == str(service_nr):
+                        _logger.info("service number %s found" % service_nr)
+                        contract_id = contract.find(".//ns62:avtalId", namespaces={'ns62':'http://arbetsformedlingen.se/datatyp/tjansteleverantor/avtal/v15'})
+                        if contract_id is not None:
```

Comment:
Same here, is it important that its not None? or can this be replaced with if contract_id